### ISSUE TITLE
undefined method `get_request_token'

### ISSUE THREAD

**Issue by roshini1972** [xero-gateway/xero_gateway#73]
Hi

Generated the Partner App specific certificates and when i tried to Authorize using xero_gateway, i am getting the following error

/.rvm/gems/ruby-1.9.3-p484/gems/xero_gateway-2.1.0/lib/xero_gateway/oauth.rb:48:in `request_token': undefined method `get_request_token' for nil:NilClass (NoMethodError)

This is the code snippet

path_to_private_key = File.join("privatekey.pem");
path_to_ssl_client_cert = File.join("entrust-cert.pem")
path_to_ssl_client_key = File.join("entrust-private-nopass.pem")
consumer_key = "XXXX"
consumer_secret = "XXXXX"

gateway = XeroGateway::PartnerApp.new(consumer_key,consumer_secret,:ssl_client_cert  => path_to_ssl_client_cert,:ssl_client_key => path_to_ssl_client_key,:private_key_file => path_to_private_key)
request_token = gateway.request_token(:oauth_callback => 'http://xxx.ngrok.com')

Let me know, if am missing out something

Thanks
Roshini

**Comment by nikz** [comment]
Hi Roshini,

It seems like your error backtrace is coming from line 48 in lib/xero_gateway/oauth.rb.

In version 2.1.0 of the gem, that line appears blank:

https://github.com/xero-gateway/xero_gateway/blob/2.1.0/lib/xero_gateway/oauth.rb#L48

Can you `bundle open` the gem and paste me the contents around the line which the error is coming from?

Thanks,

Nik

**Comment by roshini1972** [comment]
Hi Nik

The problem is resolved. The issue was due to  Public and Private key . I regenerated once again, and it worked fine. 
btw, edited oauth.rb for debugging purpose. Thats the reason for the mismatch. 

Now i am stuck with renew_access_token. I get 401 Unauthorized error.

This is what i tried 
gateway.renew_access_token(gateway.access_token.token, gateway.access_token.secret, gateway.session_handle) 

gateway.session_handle is nil in my  case. 

Thanks & Regards
Roshini
